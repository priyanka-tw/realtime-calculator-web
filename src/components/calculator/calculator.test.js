import React from 'react';
import {act, fireEvent, render} from '@testing-library/react';
import 'jest-styled-components';
import Calculator from "./calculator";

describe('Calculator', () => {

    beforeEach(() => {
        fetch.resetMocks();
    });

    it('should render the component', () => {
        const {container} = render(<Calculator/>);

        expect(container).toMatchSnapshot();
    });

    it('should display the clicked digits on screen', async () => {
        const {getByTestId} = render(<Calculator/>);

        await act(async () => {
            await fireEvent.click(getByTestId('calc-button-1'));
        });
        await act(async () => {
            await fireEvent.click(getByTestId('calc-button-3'));
        });

        expect(getByTestId('output-screen').value).toStrictEqual('13');
    });

    it('should reset the screen on click of `AC`', async () => {
        const {getByTestId} = render(<Calculator/>);

        await act(async () => {
            await fireEvent.click(getByTestId('calc-button-1'));
            await fireEvent.click(getByTestId('calc-button-3'));
        });
        expect(getByTestId('output-screen').value).toStrictEqual('13');

        await act(async () => {
            await fireEvent.click(getByTestId('calc-button-AC'));
        });
        expect(getByTestId('output-screen').value).toStrictEqual('');
    });

    it('should backspace by one on click of `C`', async () => {
        const {getByTestId} = render(<Calculator/>);

        await act(async () => {
            await fireEvent.click(getByTestId('calc-button-1'));
            await fireEvent.click(getByTestId('calc-button-3'));
        });
        expect(getByTestId('output-screen').value).toStrictEqual('13');

        await act(async () => {
            await fireEvent.click(getByTestId('calc-button-C'));
        });
        expect(getByTestId('output-screen').value).toStrictEqual('1');
    });

    it('should make api call to calculate expression on click of `=`', async () => {
        fetch.mockResponseOnce(JSON.stringify({expression: "13+5", result: "18"}));
        const {getByTestId} = render(<Calculator/>);

        await act(async () => {
            await fireEvent.click(getByTestId('calc-button-1'));
            await fireEvent.click(getByTestId('calc-button-3'));
            await fireEvent.click(getByTestId('calc-button-+'));
            await fireEvent.click(getByTestId('calc-button-5'));
        });

        await act(async () => {
            await fireEvent.click(getByTestId('calc-button-='));
        });

        expect(getByTestId('output-screen').value).toStrictEqual('18');
    });

});
