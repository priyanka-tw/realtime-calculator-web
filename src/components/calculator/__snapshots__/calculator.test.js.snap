// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render the component 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
}

.c5 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
  color: #ffc107;
}

.c6 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
  min-width: 120px;
}

.c7 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
  background-color: #ffc107;
}

.c1 {
  color: black;
  font-size: 16pt;
  font-family: verdana-bold;
  background-color: #A9A9A9;
  border: none;
  outline: none;
  text-align: right;
  width: 236px;
  height: 60px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 50px;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      data-testid="output-screen"
      readonly=""
      value=""
    />
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <button
          class="c4"
          data-testid="calc-button-AC"
          name="AC"
          type="button"
        >
          AC
        </button>
        <button
          class="c4"
          data-testid="calc-button-C"
          name="C"
          type="button"
        >
          C
        </button>
        <button
          class="c4"
          data-testid="calc-button-%"
          name="%"
          type="button"
        >
          %
        </button>
        <button
          class="c5"
          data-testid="calc-button-/"
          name="/"
          type="button"
        >
          /
        </button>
      </div>
      <div
        class="c3"
      >
        <button
          class="c4"
          data-testid="calc-button-7"
          name="7"
          type="button"
        >
          7
        </button>
        <button
          class="c4"
          data-testid="calc-button-8"
          name="8"
          type="button"
        >
          8
        </button>
        <button
          class="c4"
          data-testid="calc-button-9"
          name="9"
          type="button"
        >
          9
        </button>
        <button
          class="c5"
          data-testid="calc-button-*"
          name="*"
          type="button"
        >
          *
        </button>
      </div>
      <div
        class="c3"
      >
        <button
          class="c4"
          data-testid="calc-button-4"
          name="4"
          type="button"
        >
          4
        </button>
        <button
          class="c4"
          data-testid="calc-button-5"
          name="5"
          type="button"
        >
          5
        </button>
        <button
          class="c4"
          data-testid="calc-button-6"
          name="6"
          type="button"
        >
          6
        </button>
        <button
          class="c5"
          data-testid="calc-button--"
          name="-"
          type="button"
        >
          -
        </button>
      </div>
      <div
        class="c3"
      >
        <button
          class="c4"
          data-testid="calc-button-1"
          name="1"
          type="button"
        >
          1
        </button>
        <button
          class="c4"
          data-testid="calc-button-2"
          name="2"
          type="button"
        >
          2
        </button>
        <button
          class="c4"
          data-testid="calc-button-3"
          name="3"
          type="button"
        >
          3
        </button>
        <button
          class="c5"
          data-testid="calc-button-+"
          name="+"
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="c3"
      >
        <button
          class="c6"
          data-testid="calc-button-0"
          name="0"
          type="button"
        >
          0
        </button>
        <button
          class="c4"
          data-testid="calc-button-."
          name="."
          type="button"
        >
          .
        </button>
        <button
          class="c7"
          data-testid="calc-button-="
          name="="
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
